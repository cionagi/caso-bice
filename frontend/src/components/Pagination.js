/* eslint-disable react-hooks/exhaustive-deps */
// Dependency
import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';

// Components

const Pagination = ({ current, totalPage, handlePagination }) => {
  const [items, setItems] = useState([]);

  const drawItemPagination = () => {
    const itemElement = [];
    // eslint-disable-next-line no-plusplus
    for (let i = 0; i < totalPage; i++) {
      itemElement.push(
        <li className={`page-item ${i === current && 'active'}`} aria-current="page" key={i}>
          <span
            tabIndex={i}
            role="button"
            className="page-link "
            onClick={() => {
              handlePagination(i);
            }}
            onKeyDown={() => {
              handlePagination(i);
            }}
            data-testid={`pagination-item-${i}`}
          >
            {i + 1}
          </span>
        </li>
      );
    }
    setItems(itemElement);
  };

  useEffect(() => {
    if (totalPage) {
      drawItemPagination();
    }
  }, [current, totalPage]);

  return (
    <nav>
      <ul className="pagination pagination-sm" data-testid="pagination">
        {items}
      </ul>
    </nav>
  );
};

Pagination.defaultProps = {
  current: 1,
  totalPage: 1,
};

Pagination.propTypes = {
  current: PropTypes.number,
  totalPage: PropTypes.number,
  handlePagination: PropTypes.func,
};

export default Pagination;
