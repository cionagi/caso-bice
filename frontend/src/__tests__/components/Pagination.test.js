import React from 'react';
import { render } from '@testing-library/react';

import Pagination from '../../components/Pagination';

test('Link renders correctly', () => {
  const { asFragment } = render(<Pagination />);
  expect(asFragment()).toMatchSnapshot();
});
