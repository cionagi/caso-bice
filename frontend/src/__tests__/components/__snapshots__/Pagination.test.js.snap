// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link renders correctly 1`] = `
<DocumentFragment>
  <nav>
    <ul
      class="pagination pagination-sm"
    >
      <li
        aria-current="page"
        class="page-item false"
      >
        <span
          class="page-link "
          role="button"
          tabindex="0"
        >
          1
        </span>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;
